builtin/providers/aws/resource_aws_db_instance.go

```diff
@@ -605,10 +611,10 @@ func resourceAwsDbInstanceDelete(d *schema.ResourceData, meta interface{}) error
 
 	opts := rds.DeleteDBInstanceInput{DBInstanceIdentifier: aws.String(d.Id())}
 
-	finalSnapshot := d.Get("final_snapshot_identifier").(string)
-	if finalSnapshot == "" {
-		opts.SkipFinalSnapshot = aws.Bool(true)
-	} else {
+	skipFinalSnapshot := d.Get("skip_final_snapshot").(bool)
+	opts.SkipFinalSnapshot = aws.Bool(skipFinalSnapshot)
+	if !skipFinalSnapshot && d.Get("final_snapshot_identifier").(string) != "" {
```

Comment:
Unverified but I think you might be able to use `GetOk` here instead of `Get` and avoid the check.